### PR TITLE
FEA: add MultiDAE, MultiVAE and AutoEncoderUserDataloader for auto-encoder recommender

### PR DESCRIPTION
**Pull request by guoyihonggyh**

### CODE REVIEW

File: recbole/model/general_recommender/multidae.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        self.decode_layer_dims = [self.lat_dim] + self.encode_layer_dims[::-1][1:]
+
+        self.encoder = self.mlp_layars(self.encode_layer_dims)
+        self.decoder = self.mlp_layars(self.decode_layer_dims)
+
+        # parameters initialization
+        self.apply(xavier_normal_initialization)
+
+    def get_rating_matrix(self, user_id):
+        return torch.tensor(self.interaction_matrix[user_id.cpu(), :].toarray(), device=self.device)
```
Comment (linzihan-backforward) [inline recbole/model/general_recommender/multidae.py:51]:
```suggestion
        return torch.tensor(self.interaction_matrix[user_id.cpu(), :].toarray(), device=self.device)
```
I sugguest not to get a batch of user's interaction vector in this way. There is a more efficient way that you could refer to the implement of   DMF. 
https://github.com/RUCAIBox/RecBole/blob/511cda89deca404808bea1df2f5e98e17e9015f1/recbole/model/general_recommender/dmf.py#L141-L158

File: recbole/model/general_recommender/multidae.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    def mlp_layars(self, layer_dims):
```
Comment (linzihan-backforward) [inline recbole/model/general_recommender/multidae.py:53]: MLP layer is already built in here:
https://github.com/RUCAIBox/RecBole/blob/511cda89deca404808bea1df2f5e98e17e9015f1/recbole/model/layers.py#L29

Comment (guoyihonggyh) [reply recbole/model/general_recommender/multidae.py:65]: We don't need to apply apply activation function to the last layer of Encoder of MultiVAE and decoder of both while it seems that the MLPLayers apply activation function to every layer.

**Comment by guoyihonggyh** [comment]
There are few changes on my repository and I have create a new pull request. See https://github.com/RUCAIBox/RecBole/pull/626

**Pull request closed without merging**